swagger/smart_transfer_protocol_ws.yaml

```diff
@@ -0,0 +1,332 @@
+        contentType: application/json
+        payload:
+          type: object
+          description: Empty object, indicating succesfull completion
+
+  dataset/push/complete/confirmed:
+    publish:
+      operationId: odf.dataset.push.complete.confirmed
+      summary: Informs consumer the push completion is confirmed
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetPushCompleteConfirmed
+        name: DatasetPushCompleteConfirmed
+        contentType: application/json
+        payload:
+          type: object
+          description: Empty object, indicating succesfull completion confirmation
+
+  dataset/error:
+    publish:
+      operationId: odf.dataset.error
+      summary: Informs about error happened during one of API operations with the dataset
+      security:
+        - jwt: []
+      message:
+        messageId: odf.dataset.DatasetError
+        name: DatasetError
+        contentType: application/json
+        payload:
+          type: object
+          required:
+            - errorDetails
+          properties:
+            errorDetails:
+              $ref: '#/components/schemas/ErrorDetails'
+
+components:
+  schemas:
+    ObjectsBatch:
+      type: object
+      description: An archive of blocks segment being transferred
+      required:
+        - objectsCount
+        - objectType
+        - mediaType
+        - encoding
+        - payload
+      properties:
+        objectsCount:
+          type: integer
+          minimum: 0
+          description: Number of objects included in the batch
+        objectType:
+          type: string
+          example: "MetadataBlock"
+          description: Type of objects included in the batch
+        mediaType:
+          type: string
+          example: "application/tar"
+          description: Payload media type
+        encoding:
+          type: string
+          example: base64
+          description: Payload encoding within the protocol
+        payload:
+          type: string
+          description: Encoded payload data
+
+    NegotiatedDownload:
+      type: object
+      required:
+        - objectFile
+        - downloadFrom
+      properties:
+        objectFile:
+          $ref: '#/components/schemas/ObjectFileReference'
+        downloadFrom:
+          $ref: '#/components/schemas/StorageReference'    
+
+    NegotiatedUpload:
+      type: object
+      required:
+        - objectFile
+        - uploadNegotiationStatus
+      properties:
+        objectFile:
+          $ref: '#/components/schemas/ObjectFileReference'
+        uploadNegotiationStatus:
+          type: string
+          enum: [ExpectingUpload, SkipUpload]
+        uploadTo:
+          $ref: '#/components/schemas/StorageReference'    
+
+    ObjectFileReference:
+      type: object
+      required:
+        - fileKind
+        - physicalHash
+      properties:
+        fileKind:
+          $ref: '#/components/schemas/ObjectFileKind'
```

Comment:
Can we unify `ObjectsBatch.objectType` and `ObjectFileReference.fileKind` fields?

Their meaning seems to be the same, so I think it would be better if they were both named `objectType` and used `ObjectType` enum type (that would replace `ObjectFileKind`).

Reply:
Okay. This could also open a way to send a tarball of checkpoints and data slcies too, if necessary. Maybe it's a good option for little datasets.